### PR TITLE
Fix logging for different users

### PR DESCRIPTION
**Pull request by NataMarkova**
@jrmcgarvey, please check it out. I completed "require logon for some operations only" and "FIx services index view so that the edit and delete buttons do not show if no one is logged in." cards

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
-  # before_action :authenticate_user!
+  before_action :authenticate_user!
```
Comment (jrmcgarvey) [inline app/controllers/application_controller.rb:3]: This line should not be here -- else you must always log in, even to see the welcome page. I don't really see how you could test the rest of your changes if this line is here.

File: app/views/services/index.html.erb [hunk 2]
```diff
@@ -61,9 +61,13 @@
           <% end %>
           </figure>
         </td>
-        <td><%= link_to 'Show', service %></td>
-        <td><%= link_to 'Edit', edit_service_path(service) %></td>
-        <td><%= link_to 'Destroy', service, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% if current_user === nil %>
```
Comment (jrmcgarvey) [inline app/views/services/index.html.erb:64]: This is basically correct, but you have to fix the colspan for the table as well -- you are doing colspan 3 for the show edit destroy links even when you don't have edit/destroy.  The change is about line 41.

File: app/views/videos/index.html.erb [hunk 3]
```diff
@@ -18,10 +18,14 @@
         <td><%= video.title %></td>
         <td><%= video.description %></td>
         <td><%= video.text %></td>
-        <td><%= link_to 'Show', video %></td>
-        <td><%= link_to 'Edit', edit_video_path(video) %></td>
-        <td><%= link_to 'Destroy', video, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% if current_user === nil %>
```
Comment (jrmcgarvey) [inline app/views/videos/index.html.erb:21]: Again, you have to fix the colspan above as well.

**Review by jrmcgarvey — CHANGES_REQUESTED** [review]
Some changes needed, see comments inline

File: app/views/services/index.html.erb (hunk 2 above)
Comment (NataMarkova) [reply app/views/services/index.html.erb:64]: Below line 41 should be the code that indicates a different type of users?
`<% if current_user === nil %>
        <th colspan="1">
  <% else %>
        <th colspan="3">
  <% end %>`

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (NataMarkova) [reply app/controllers/application_controller.rb:3]: Of course! I left the line accidentally when played with code changes.

**Review by jrmcgarvey — APPROVED** [review]
Looks ok now.

**Pull request merged**